core/src/main/scala/org/apache/spark/rpc/RpcEndpoint.scala

```diff
@@ -146,3 +146,13 @@ private[spark] trait RpcEndpoint {
  * [[ThreadSafeRpcEndpoint]] for different messages.
  */
 private[spark] trait ThreadSafeRpcEndpoint extends RpcEndpoint
+
+/**
+ * An endpoint that uses a dedicated thread pool for delivering messages.
+ */
+private[spark] trait IsolatedRpcEndpoint extends RpcEndpoint {
+
+  /** How many threads to use for delivering messages. By default, use a single thread. */
+  def threadCount(): Int = 1
```

Comment:
I'm trying to wrap my head around what happens if you create an `IsolatedRpcEndpoint` with threadCount() > 1, given the code in `Inbox` which checks for inheritance from `ThreadSafeRpcEndpoint`: https://github.com/apache/spark/blob/2b3c3793c97dc0e48f1e098c6f1038f51607b716/core/src/main/scala/org/apache/spark/rpc/netty/Inbox.scala#L123

I guess if you expect one endpoint to be served by multiple threads, it makes sense you'd want `Inbox.enableConcurrent = false` and you'd have to make your endpoint safe for that -- but worth a comment here at least.

Reply:
I have the same question with @squito . How do you deal with `ThreadSafeRpcEndpoint` ?

Though we could set `Inbox.enableConcurrent = false` with `threadCount() > 0`, but multiple threads would be wasted.

Reply:
I already updated the comment. `ThreadSafeRpcEndpoint` is irrelevant here. You may even extend both if you want; but if you do that, either it does nothing (because the thread pool has a single thread), or you're doing it wrong (because the thread pool has multiple thread but you just want one).

So it's pointless to mix in both traits.